src/umpire/resource/FileMemoryResource.hpp

```diff
@@ -0,0 +1,45 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#ifndef UMPIRE_FileMemoryResource_HPP
+#define UMPIRE_FileMemoryResource_HPP
+
+#include "umpire/resource/MemoryResource.hpp"
+
+#include "umpire/util/Platform.hpp"
+#include "umpire/util/MemoryMap.hpp"
+
+#include <utility>
+
+namespace umpire {
+namespace resource {
+
+class FileMemoryResource :
+  public MemoryResource
+{
+  public: 
+    FileMemoryResource(Platform platform, const std::string& name, int id, MemoryResourceTraits traits);
+
+    void* allocate(std::size_t bytes);
+    void deallocate(void* ptr);
+
+    std::size_t getCurrentSize() const noexcept;
+    std::size_t getHighWatermark() const noexcept;
+
+    Platform getPlatform() noexcept;
+
+  protected:
+    Platform m_platform;
+
+  private:
+    util::MemoryMap<std::pair <const std::string, std::size_t>> m_size_map;
+    int FILE_COUNTER = 0;
```

Comment:
```suggestion
    static int s_file_counter{0};
```

Reply:
If we make this static we get this error v but if left as int the code runs fine
`[ 28%] Building CXX object src/umpire/resource/CMakeFiles/umpire_resource.dir/FileMemoryResource.cpp.o
In file included from /home/aileen/Umpire/src/umpire/resource/FileMemoryResource.cpp:9:
/home/aileen/Umpire/src/umpire/resource/FileMemoryResource.hpp:39:16: error: ISO C++ forbids in-class initialization of non-const static member ‘umpire::resource::FileMemoryResource::s_file_counter’
   39 |     static int s_file_counter{0};
      |                ^~~~~~~~~~~~~~
`

Reply:
It needs to be `static` so that it will be shared across all object instantiations in the process.  Since this is not a `const`, I think that this needs to be initialized in `FileMemoryResource.cpp`.